6.0-Upgrade.md

```diff
@@ -45,7 +45,16 @@ Check the following list to see if you're depending on any of these behaviors:
 1. We've removed the following constants: `Puma::StateFile::FIELDS`, `Puma::CLI::KEYS_NOT_TO_PERSIST_IN_STATE` and `Puma::Launcher::KEYS_NOT_TO_PERSIST_IN_STATE`, and `Puma::ControlCLI::COMMANDS`.
 1. We no longer support Ruby 2.2, 2.3, or JRuby on Java 1.7 or below.
 1. The behavior of `remote_addr` has changed. When using the set_remote_address header: "header_name" functionality, if the header is not passed, REMOTE_ADDR is now set to the physical peeraddr instead of always being set to 127.0.0.1. When an error occurs preventing the physical peeraddr from being fetched, REMOTE_ADDR is now set to the unspecified source address ('0.0.0.0') instead of to '127.0.0.1'
-1. Previously, Puma supported anything as an HTTP method and passed it to the app. We now only accept the 8 HTTP methods defined in RFC 9110.
+1. Previously, Puma supported anything as an HTTP method and passed it to the app. We now only accept the 8 HTTP methods defined in RFC 9110. If you want to override which HTTP methods are accepted, you may use the `supported_http_methods` option:
+    ```ruby
+      supported_http_methods %w[
+        ACL BASELINE-CONTROL BIND CHECKIN CHECKOUT CONNECT COPY DELETE
+        GET HEAD LABEL LINK LOCK MERGE MKACTIVITY MKCALENDAR MKCOL
+        MKREDIRECTREF MKWORKSPACE MOVE OPTIONS ORDERPATCH PATCH POST
+        PRI PROPFIND PROPPATCH PUT REBIND REPORT SEARCH TRACE UNBIND
+        UNCHECKOUT UNLINK UNLOCK UPDATE UPDATEREDIRECTREF VERSION-CONTROL
+      ]
```

Comment:
So with this new option you can make Puma not to respond to the most common HTTP methods, may be a good feature for someone, but also a bit of a sharp knife if you forget to include a method. Docs could state something along the lines "make sure you include all the HTTP methods your app support here if you use this option"